jme3-core/src/main/java/com/jme3/animation/SkeletonControl.java

```diff
@@ -443,9 +459,19 @@ public Node getAttachmentsNode(String boneName) {
                     + "in the skeleton.");
         }
 
-        Node n = b.getAttachmentsNode();
-        Node model = (Node) spatial;
-        model.attachChild(n);
+        updateTargetsAndMaterials(spatial);
+        Node n = b.getAttachmentsNode(targetGeometry);
```

Comment:
IMO that's a wrong assumption.
What prevents a user from having several geometries with different transforms?
I mean this is the very issue we are trying to fix. We assumes geoms wouldn't have transforms, now we assume they have the same transforms. It looks like the same mistake to me.

Also you're right about the second statement... A bone can affects multiple geoms. Making the assumption in this particular case, and picking the first/last one, is less likely to cause trouble. Thought it doesn't justify the fact that it's wrong when a bone affects only one geom.